src/library/FIFO.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A First-In-First-Out (FIFO) class that maintains a queue of
+ * elements.  New elements are added to the tail, and the oldest
+ * element is removed from the head when the FIFO is full.  If the
+ * size of the queue is 1, it will always contain the newest element
+ * that was inserted.
+ * @class
+ */
+
+class FIFO {
+  
+  /**
+   * @constructor Constructs a FIFO (First-In-First-Out) object.
```

Comment:
Remove "Constrcuts a FIFO ....". We only need @constructor tag because you already provided an overview above.